### ISSUE TITLE
Autowrap line-breaks incorrectly at a Thai word containing upper/lower vowels or tone marks

### ISSUE THREAD

**Issue by Liely118** [godotengine/godot#99474]
### Tested versions

Reproducible in: 4.3.stable, 4.4.dev4

### System information

Godot v4.4.dev4 - Windows 10.0.19045 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1660 SUPER (NVIDIA; 31.0.15.3713) - AMD Ryzen 5 3600 6-Core Processor (12 threads)

### Issue description

With Autowrap mode set to Word (or Word (Smart)), when a line is supposed to break in the middle of a Thai word that contains upper/lower vowels or tone marks, it displays the consonant with those characters twice, one each before and after the line break. However, in some specific cases, instead of displaying a few characters twice, an entire word is not displayed at all.

Below is an example of incorrectly displayed line breaks. The spaces before the text have no effects on the broken line breaks.
The Correct column is me manually adding line breaks to show what it should've look like.

![Image](https://github.com/user-attachments/assets/0315c357-3e3b-4ae6-a698-4801cd8c15b8)

Here are the problematic characters:
(Taken from https://jrgraphix.net/r/Unicode/0E00-0E7F)

![Image](https://github.com/user-attachments/assets/82083f26-c05b-47df-b493-297d670b02a7)

### Steps to reproduce

- Add a Label
- Set Autowrap mode to Word
- Set text to "             กู้ไฟล์"
- Adjust Label size until a line break occurs at the word "กู้"

### Minimal reproduction project (MRP)

Project containing the table in the screenshot: [ThaiAutowrapBug.zip](https://github.com/user-attachments/files/17833870/ThaiAutowrapBug.zip)

**Comment by AThousandShips** [comment]
CC @bruvzg

**Comment by bruvzg** [comment]
![Image](https://github.com/user-attachments/assets/39148da5-ea0b-460d-83e6-791ea0441727)

Not reproducible on macOS, since project doesn't include any fonts, it might be an issue with specific font (whatever is installed on the specific system), will check in on Windows later.

To check which font is actually used, please add the following script to any project, run it on your system and share the output:
```gdscript
func _ready() -> void:
	print(OS.get_system_font_path_for_text("sans", "กู้ไฟล์"))
```

**Comment by Liely118** [comment]
The font is "C:/WINDOWS/FONTS/LeelawUI.ttf". But I actually noticed this issue while working on my project, which uses Google's NotoSansThai font.

Using other fonts and readjusting the text a bit presents the same issue.

NotoSansThai font:
![Image](https://github.com/user-attachments/assets/8e039a7f-6bea-429c-a8ae-ff5dbadb2698)

Itim font:
![Image](https://github.com/user-attachments/assets/b3a2ee96-e1f7-4cbf-bdce-ae3b4235930e)

I've included 3 fonts, all from Google Fonts, that have the same problem in an updated project:
[ThaiAutowrapBug2.zip](https://github.com/user-attachments/files/17834918/ThaiAutowrapBug2.zip)
